Sources/SemanticIndex/SemanticIndexManager.swift

```diff
@@ -350,21 +352,21 @@ public final actor SemanticIndexManager {
       switch newState {
       case .executing:
         for file in files {
-          if case .scheduled((taskID, let task)) = self.indexStatus[file.uri] {
-            self.indexStatus[file.uri] = .executing((taskID, task))
+          if case .scheduled((taskID, let task)) = self.indexStatus[file.file.sourceFile] {
```

Comment:
`file.file.sourceFile` is... amusing 😅. I don't have a great suggestion here outside of naming `sourceFile` -> `sourceUri` and then maybe the variable `file` -> `fileWithTarget`. Or just leave them but add a `let sourceFile = file.file.sourceFile` so we don't have the many references 🤷

Reply:
Changed it to `fileAndTarget.file.sourceFile`. It’s at least a little better.